### PR TITLE
Fix post processor skip test case if return 0

### PR DESCRIPTION
**Pull request by 1984gobrr**
https://github.com/AFLplusplus/AFLplusplus/issues/944

### CODE REVIEW

File: src/afl-fuzz-run.c [hunk 1]
```diff
@@ -205,14 +205,17 @@ static void write_with_gap(afl_state_t *afl, u8 *mem, u32 len, u32 skip_at,
         new_size =
             el->afl_custom_post_process(el->data, new_mem, new_size, &new_buf);
 
-        if (unlikely(!new_buf || new_size <= 0)) {
+        if (unlikely(!new_buf)) { FATAL("Custom_post_process failed"); }
```
Comment (vanhauser-thc) [inline src/afl-fuzz-run.c:208]: I would keep this as is and just check for "new_size < 0" instead of "<=".

File: src/afl-fuzz-run.c [hunk 2]
```diff
@@ -205,14 +205,17 @@ static void write_with_gap(afl_state_t *afl, u8 *mem, u32 len, u32 skip_at,
...
 
-          FATAL("Custom_post_process failed (ret: %lu)",
-                (long unsigned)new_size);
+        if (unlikely(new_size <= 0)) {
 
-        }
+          LIST_REMOVE_CURRENT_EL_IN_FOREACH()
```
Comment (vanhauser-thc) [inline src/afl-fuzz-run.c:212]: why remove? and why remove if the size is 0 (which is a legal case, that is what this PR should be about)

**Comment by vanhauser-thc** [comment]
beside the two comments I left another thing that is missing is that afl-fuzz has to be instructed to skip the testcase if the size is 0. this is what this PR has to be about.
thank you for the effort so far.

**Comment by 1984gobrr** [comment]
Hummm crap, I misunderstood the issue.
I will make an other PR to try to fix it, sorry for that :face_in_clouds:

**Pull request closed without merging**